### PR TITLE
feature5 added - user form

### PR DESCRIPTION
**Pull request by krzysiekpom**
In the admin panel there is a form, where an email, gender, age, interests can be specified for new and existing users. Requirement: the form should offer a possibility to dynamically specify the interest for the user, by adding additional fields to the form, without reloading the page. Each interest can have a name and a type (from the predefined collection of: ["health", "hobby", "work"]). When it comes to the number of interests, user can have any amount. New users created by admin should have a password set to "secret". Hint: https://github.com/ryanb/nested_form or https://github.com/nathanvda/cocoon

Like you said, feature3 and feature4 integrated automatically. Feels good :)

### CODE REVIEW

File: app/views/admin/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          <%= render 'interest_fields', f: interest %>
+        <% end %>
+      </tbody>
+    </table>    
+
+    <%= f.submit 'Save' %>
+  </div>
+<% end %>
+
+
```
Comment (socjopata) [inline app/views/admin/_form.html.erb]: ✂️

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module UsersHelper
+  def all_users_filter_list
+    User.all.map do |user| 
+      [user.email, user.id] 
+    end.uniq.sort
+  end
+
+  def interest_available_options
+    %w(health hobby work)
```
Comment (socjopata) [inline app/helpers/users_helper.rb]: Please define this in a model. Add validations

File: app/views/admin/_interest_fields.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<tr class='nested-fields'>
+  <td><%= f.select(:interest_type, options_for_select(interest_available_options, selected: f.object.interest_type)) %></td>
+  <td><%= f.text_field :name %></td>
+  <td><%= link_to_remove_association 'REMOVE INTEREST', f %></td>
+</tr>
+
```
Comment (socjopata) [inline app/views/admin/_interest_fields.html.erb]: ✂️

File: app/helpers/users_helper.rb (hunk 2 above)
Comment (krzysiekpom) [reply app/helpers/users_helper.rb]: I extracted the types in the Interest model into a constant and used it in validations as well as in interest_fields partial (in place of the interest_available_options helper).

**Pull request merged**